src/formatted-codegen.js

```diff
@@ -542,14 +546,22 @@ export class ExtensibleCodeGen {
     return this.bracket(content, Sep.ARRAY_INITIAL, Sep.ARRAY_FINAL, Sep.ARRAY_EMPTY);
   }
 
-  reduceObjectAssignmentTarget(node, { properties }) {
-    let state = this.brace(this.commaSep(properties, Sep.OBJECT_BEFORE_COMMA, Sep.OBJECT_AFTER_COMMA), node, Sep.OBJECT_BRACE_INITIAL, Sep.OBJECT_BRACE_FINAL, Sep.OBJECT_EMPTY);
+  reduceObjectAssignmentTarget(node, { properties, rest }) {
+    let content = this.commaSep(properties, Sep.OBJECT_BEFORE_COMMA, Sep.OBJECT_AFTER_COMMA);
+    if (rest != null) {
+      content = seq(content, this.sep(Sep.OBJECT_BEFORE_COMMA), this.t(','), this.sep(Sep.OBJECT_AFTER_COMMA), this.t('...'), this.sep(Sep.REST), rest);
```

Comment:
If there are no properties other than the rest, this will codegen `({ ...rest} = 0)` as `({ , ...rest} = 0)`, which is invalid. You could either manually check if the `properties` array is of length 0, or use `commaSep`, as in

```js
content = this.commaSep([content, seq(this.t('...'), this.sep(Sep.REST), rest)], Sep.OBJECT_BEFORE_COMMA, Sep.OBJECT_AFTER_COMMA);
```

Should add this case as a test.

Same comment on object bindings.